### PR TITLE
Show line number and assertion when verifying generated markdown files in lines

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Show line number and assertion when verifying generated markdown files, e.g. `settings.md` from `AllKyuubiConfiguration`, for quicker locating the place and differences in files.
- updated hint message to `mvn clean test` instead of `mvn clean install` for faster verification or regeneration

Hints with line num as below if assertion fails in line comparison.
image

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/test/scala/org/apache/kyuubi/TestUtils.scala [hunk 1]
```diff
@@ -92,11 +92,13 @@ object TestUtils {
...
       val formatted = formatMarkdown(newOutput)
-      val hint = s"$markdown out of date, please update doc with " +
-        s"KYUUBI_UPDATE=1 build/mvn clean install -Pflink-provided,spark-provided,hive-provided " +
-        s"-DwildcardSuites=$agent"
-      assert(linesInFile.size() === formatted.size, hint)
-      linesInFile.asScala.zip(formatted).foreach { case (out, in) => assert(out === in, hint) }
+      linesInFile.asScala.zipWithIndex.zip(formatted).foreach { case ((str1, index), str2) =>
+        withClue(s"$markdown out of date, as line ${index + 1} is not expected." +
+          s" Please update doc with KYUUBI_UPDATE=1 build/mvn clean install" +
+          s" -Pflink-provided,spark-provided,hive-provided -DwildcardSuites=$agent") {
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/TestUtils.scala:98]: The following command should be sufficient and faster
```
KYUUBI_UPDATE=1 build/mvn clean test -Pflink-provided,spark-provided,hive-provided  -pl kyuubi-server -am -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration
```

Comment (bowenliang123) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/TestUtils.scala:98]: Addressed.

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/TestUtils.scala:98]: Please search "AllKyuubiConfiguration" globally and make sure all places are updated.

Comment (bowenliang123) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/TestUtils.scala:98]: OK.

File: docs/develop_tools/developer.md [hunk 2]
```diff
@@ -56,5 +56,5 @@ You can run `dev/reformat` to format all Java and Scala code.
 
 Kyuubi uses settings.md to explain available configurations.
 
-You can run `KYUUBI_UPDATE=1 build/mvn clean install -Pflink-provided,spark-provided,hive-provided -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration`
+You can run `KYUUBI_UPDATE=1 build/mvn clean test -Pflink-provided,spark-provided,hive-provided -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration`
```
Comment (pan3793) [inline docs/develop_tools/developer.md:59]: it takes more time w/o `-pl kyuubi-server -am`.

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**